__notes__/3.1-Configuration_and_Environment.md

```diff
@@ -0,0 +1,79 @@
+# Configuration And Environment
+
+### Add 2 more dependencies to the application
+
+1. config
+
+- https://www.npmjs.com/package/config
+
+```sh
+npm install config
+```
+
+2. cross-env
+
+- https://www.npmjs.com/package/cross-env
+- cross-platform environment
+
+```sh
+npm install --save-dev cross-env
+```
+
+3. configure scripts in package.json
+
+```json
+  "scripts": {
+    "start": "node build/index.js",
+    "dev": "cross-env NODE_ENV=development nodemon --watch './**/*.ts' --exec 'ts-node' index.ts",
+    "test": "cross-env NODE_ENV=test jest --watch",
+    "lint": "eslint ."
+  },
+```
+
+4. add config folder in the root, and add these config files depending on the NODE_ENV
+
+```json
+// config/development./json
+{
+  "database": "hoaxify",
+  "username": "my-db-user",
+  "password": "db-p4ss",
+  "dialect": "sqlite",
+  "storage": "./database.sqlite",
+  "logging": false
+}
+
+// config/test.json
+{
+  "database": "hoaxify",
+  "username": "my-db-user",
+  "password": "db-p4ss",
+  "dialect": "sqlite",
+  "storage": ":memory:", // more suitable for test, rather than ./test.sqlite
```

Comment:
do "storage": "./test.sqlite" if you don't want it to be in the memory, more preferable to store into memory if only testing. 

only in sqlite